dockstore/dockstore-support
# Add command to submit validation data & aggregate validation status

[SEAB-4695](https://ucsc-cgl.atlassian.net/browse/SEAB-4695)

View https://github.com/dockstore/dockstore/pull/5447 for the webservice changes

This PR adds a command to the metrics aggregator to submit validation data so we can add DNAstack's validation information to Dockstore. The command takes in a CSV field containing the TRS ID, version name, isValid boolean, and the date the validation was executed. I created a script that transforms the file DNAstack gave us into a CSV file. I ran the `submit-validation-data` command with the CSV file and it took about 3-4 minutes.

This PR also aggregates the validation execution data found in S3.

**Review Instructions**
- SSM into the deployer
- git clone the dockstore-support repo and do cd metricsaggregator && mvn clean install -DskipTests
- Create a configuration file as defined in the metricsaggregator README. Do not provide an S3 endpoint override. Example:
   ```
  [dockstore]
  server-url: https://qa.dockstore.org/api
  token: <Dockstore token of a curator or admin>
  
  [s3]
  bucketName: qa-dockstore-metrics-data
   ```
- Download the `workflow_names.txt` file in the ticket and put this on the Deployer. This contains workflows that validated correctly with miniwdl.
- Locate the JAR and run the java program to submit successful validation data for miniwdl. Example instructions are in the README. It should not fail.
- Run the java program to aggregate the metrics. It should not fail. Verify that there are validation metrics for some of the workflows aggregated.

[SEAB-4695]: https://ucsc-cgl.atlassian.net/browse/SEAB-4695?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
